Datastore/src/Query/AggregationQueryResult.php

```diff
@@ -0,0 +1,96 @@
+<?php
+/**
+ * Copyright 2023 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Cloud\Datastore\Query;
+
+use Google\Cloud\Core\Timestamp;
+use InvalidArgumentException;
+
+/**
+ * Represents the result of an [Aggregation Query](https://cloud.google.com/datastore/docs/aggregation-queries)
+ *
+ * Example:
+ * ```
+ * use Google\Cloud\Datastore\DatastoreClient;
+ *
+ * $datastore = new DatastoreClient();
+ *
+ * $query = $datastore->AggregationQuery();
+ * $query->kind('Companies');
+ * $query->filter('companyName', '=', 'Google');
+ * $query->limit(100);
+ * $query->addAggregation(Aggregation::count()->alias('total_upto_100'));
+ *
+ * $res = $datastore->runAggregationQuery($query);
+ * echo $res->get('total_upto_100');
+ * ```
+ *
+ * @see https://cloud.google.com/datastore/reference/rest/v1/projects/runAggregationQuery#query
+ * Aggregation Query Object Reference
+ * @see https://cloud.google.com/datastore/docs/aggregation-queries Aggregation Queries
+ */
+class AggregationQueryResult
+{
+    /**
+     * @var AggregationQuery
+     */
+    public $query;
+
+    /**
+     * @var Timestamp
+     */
+    public $readTime;
+
+    /**
+     * @var array
+     */
+    public $aggregationResults = [];
+
+    /**
+     * @var string|null
+     */
+    public $transactionId;
```

Comment:
I did a quick search through the `Datastore` code (for `    public $`), and no other public properties exist on any classes.  I did the same search for this entire repo, and outside of tests, I only found two classes with public properties (`Core\Streamwrapper::$context` and `Debugger\AliasContext::$kind` / `$name`)

My feeling is that these can be private, and if the customers need access, then we should provide getters and setters, as that's the typical API for this library and protobuf.

Reply:
Moved to a private field and added getters